src/fugw/solvers/utils.py

```diff
@@ -996,21 +1216,29 @@ def compute_quad_kl_sparse(mu, nu, alpha, beta):
 
 
 def compute_quad_l2(a, b, mu, nu):
-    """
-    Compute || a otimes b - mu otimes nu ||^2
-    """
+    """Compute || a otimes b - mu otimes nu ||^2."""
```

Comment:
This is not exactly what is computed.

Reply:
`a otimes b` and `mu otimes nu` are expensive to store in memory.
Instead, we expanded `norm2(a otimes b - mu otimes nu)` such that all manipulated quantities are scalars.
Did you find an calculus error?

Reply:
OK, otimes is tensor product.